version: '3.3'

services:
  mysql:
    image: mysql:8.0.22
    container_name: mysql
    restart: always
    environment:
      MYSQL_ROOT_PASSWORD: "password"
    ports:
      - '3306:3306'
    volumes:
#      - mysql-data:/var/lib/mysql
      - ./init:/docker-entrypoint-initdb.d/
    command:
      --character-set-server=utf8mb4
      --collation-server=utf8mb4_general_ci
      --explicit_defaults_for_timestamp=true
      --sql-mode="STRICT_TRANS_TABLES,NO_ENGINE_SUBSTITUTION,NO_ZERO_DATE,NO_ZERO_IN_DATE,ERROR_FOR_DIVISION_BY_ZERO"
    networks:
      - mynet
    healthcheck:
      test: [ "CMD", "mysqladmin" ,"ping", "-h", "localhost" ]
      timeout: 20s
      retries: 10


  redis:
    image: "redis:5.0.9-alpine"
    container_name: redis
    command:
      redis-server
    ports:
      - "6379:6379"
    volumes:
      - redis-data:/var/lib/redis
    networks:
      - mynet
    depends_on:
      mysql:
        condition: service_healthy

networks:
  mynet:
    driver: bridge

volumes:
  mysql-data:
    driver: "local"
  redis-data:
    driver: "local"
